kpops/cli/utils.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+
+from collections.abc import Generator
+from pathlib import Path
+
+
+def collect_pipeline_paths(pipeline_path: Path) -> Generator[Path, None, None]:
```

Comment:
since we're only yielding Paths without special cases I would prefer the simplified annotation

```suggestion
def collect_pipeline_paths(pipeline_path: Path) -> Iterator[Path]:
```